### PR TITLE
Add config option to disable blending

### PR DESCRIPTION
**Pull request by dmccombs**
This is intended to work around issues with some images blending incorrectly with ARGB visuals.

This is a follow up for #84 , intended as a work around for #51 and #123 .

I've tried to find a fix that works for all images, but haven't been able to. For example, this image with blending disabled on ARGB ends up with a white background instead of alpha:
tango-arch

While this image ends up being invisible entirely unless blending is disabled:
lsd-black

I ended up adding a config option so that themes can choose whether or not to blend based on their image as a way of working around this issue. I've named the config option `render_draw_blended` and it defaults to `true` to maintain how rendering is done now by default. Setting it to `false` fixes the issue with the second image above.

I'd like to hear from other users who have run into this bug on whether this helps them work around the problem, whether it helps with recent Gnome versions, and I'm open to feedback on the name of the config value and whether it should be tied to ARGB or not (it currently isn't).

### CODE REVIEW

**Comment by dmccombs** [comment]
@lasers here's an updated patch in response to your comment on #84

File: doc/config_settings.xml [hunk 1]
```diff
@@ -952,6 +952,18 @@
         <listitem>Border stippling (dashing) in pixels
         <para /></listitem>
     </varlistentry>
+    <varlistentry>
+        <term>
+            <command>
+                <option>render_draw_blended</option>
```
Comment (lasers) [inline doc/config_settings.xml]: (nit) `r` goes before `s`.

**Comment by dmccombs** [comment]
Also, to add some info on things when I was poking around for myself and others, I used the `identify` program from `imagemagick` to inspect differences between the images behaving differently. Using the `-verbose` argument, the `lsd-black.png` image shows as Grayscale, while the Archlinux logo shows as Truecolor, but adding color to the Grayscale image and saving it to where it showed as Truecolor didn't change how it behaved with Conky. I also noticed that the black image has `Alpha: none   #00000000` whereas the Arch image has `Alpha: srgba(255,255,255,0)   #FFFFFF00`, though both show up with alpha when opened in GIMP. Looking through the imlib2 library, I didn't see a way to tell things apart to handle them differently, and don't even know for sure what difference exactly is causing the issue. That's when I decided to just make this a configuration option and leave it up to the user or theme creator.

File: src/imlib2.cc [hunk 2]
```diff
@@ -61,6 +61,9 @@ conky::range_config_setting<unsigned int> imlib_cache_flush_interval(
     0, true);
 
 unsigned int cimlib_cache_flush_last = 0;
+
+conky::simple_config_setting<bool> render_draw_blended(
+    "render_draw_blended", true, true);
```
Comment (lasers) [inline src/imlib2.cc]: (question) Would `blend_images` make a better option name?

File: doc/config_settings.xml (hunk 1 above)
Comment (lasers) [reply doc/config_settings.xml]: (question) Would `blend_images` make a better option name?

Comment (dmccombs) [reply doc/config_settings.xml]: Updated the ordering, whoops.

`draw_blended` or some variation seems OK to me. At first I thought there was another blend going on that it should be separate from, but that's really just part of loading the image from the file to the buffer it seems, so maybe it doesn't need to be specific to `render`.

**Comment by dmccombs** [comment]
@lasers hmm - SonarCloud still says "Expected" here waiting for the report - do you have a link to a report you're asking about?

**Comment by lasers** [comment]
I see the edit. It's not something you changed, but what file you touched. We have this new SonarCloud in place maybe 6 months ago so each file may be hit with this over time (nice and slow).

And we _collaborators_ don't have an ability to merge PRs that does not pass the tests. That's up to _owner_ and that could take weeks. It just would be faster if you can look into this and we work together enough to pass the tests. Not sure about malloc? @npyl could look into this too.

**Comment by dmccombs** [comment]
Ah, that makes sense @lasers. If there's an example of how it was handled/changed in some other part of conky's code I'd be happy to make the change here to pass tests and keep things consistent. I'm thinking I'll probably change the name of the configuration to `draw_blended` unless anyone else has more thoughts on it as well.

**Comment by lasers** [comment]
Here's a recent example and it deals with `malloc` too: #680. You can put it in a new commit. Sure, I think `draw_blended` is better than `render_draw_blended` too. We can go with new as we can't go wrong.

**Comment by dmccombs** [comment]
Updated with a potential fix for the SonarCloud `malloc` warning, though I don't see the output of it on this PR. Also , there's a failure from Travis CI that doesn't seem related to this change:
```
0.17s$ if [ "$CXX" = "clang++-6.0" ] ; then build-wrapper-linux-x86-64 --out-dir bw-output make -j4 ; fi
build-wrapper-linux-x86-64: command not found
The command "if [ "$CXX" = "clang++-6.0" ] ; then build-wrapper-linux-x86-64 --out-dir bw-output make -j4 ; fi" exited with 127.
```

**Comment by erincar** [comment]
I was so coincidentially looking for a solution for this, just realized the release after a long time fixing this, awesome!

When will the release be available on bionic universe repositories?

**Comment by brndnmtthws** [comment]
It takes a while for these releases to get into the distros. It's fairly
easy to compile and install from source, so I suggest you give that a try.

**Comment by dmccombs** [comment]
Hmm - I've noticed that this config option works in the new config file format, but if added to the old config file format it's not being converted correctly:
```
conky --config ~/.conkyrc
conky: Syntax error (/home/dmccombs/.conkyrc:1: '=' expected near 'yes') while reading config file.
conky: Assuming it's in old syntax and attempting conversion.
conky: desktop window (16e) is root window
conky: window type - desktop
conky: drawing to created window (0xe00002)
conky: drawing to double buffer
conky: Invalid value of type 'string' for setting 'draw_blended'. Expected value of type 'boolean'.
conky: forked to background, pid is 5299
```
I tried adding it to the list of booleans in `extras/convert.lua`, but that didn't seem to do the trick. What's doing this automatic conversion on start up that knows which settings get converted to what type?

**Comment by dmccombs** [comment]
@lasers ah, yes, that did the trick. #688 incoming.

**Pull request merged**